src/middlewared/middlewared/plugins/service_/services/nfs.py

```diff
@@ -14,13 +12,17 @@ class NFSService(SimpleService):
     systemd_unit = "nfs-server"
 
     async def check_configuration(self):
-        if not await self.middleware.run_in_thread(os.path.exists, '/etc/exports'):
-            raise CallError(
-                'At least one configured and available NFS export is required '
-                'in order to start the NFS service. Check the NFS share configuration '
-                'and availability of any paths currently being exported.',
-                errno.EINVAL
-            )
+        # NAS-123498: Eliminate requirement to have shares configured to start NFS
+        # But, raise an alarm if there are entries in /etc/exports.d
+        exportsd = '/etc/exports.d'
+        if os.path.exists(exportsd) and not os.path.isfile(exportsd):
```

Comment:
This is calling blocking I/O in an async method so it'll block the event loop. Lastly, I don't like checking for existence of things before performing I/O on them if we don't have to. We should create a synchronous method and call it inside a thread. i.e.

```
def check_exportsd_dir(self):
    exports = list()
    try:
        with os.scandir('/etc/exports.d') as scan:
            for i in filter(lambda x: x.is_file() and not x.name.startswith('.'), scan):
                exports.append(i.name)
    except (FileNotFoundError, NotADirectoryError):
        pass
   return exports
   
async def check_configuration(self):
    if (exportsdList := await self.middleware.run_in_thread(self.check_exports_dir)):
        await self.middleware.call('alert.oneshot_create', 'NFSblockedByExportsDir', {'entries': exportsdList})
    else:
        await self.middleware.call('alert.oneshot_delete', 'NFSblockedByExportsDir')